### PR TITLE
update golangci-lint rules

### PR DESCRIPTION
**Pull request by jmank88**
This PR enables some new linters and revive rules for things we are already trying to do.

### CODE REVIEW

**Review by vyzaldysanchez — DISMISSED** [review]

**Comment by jmank88** [comment]
I was unable to configure `testifylint` to disable the `require-error` rule, which was producing many false positives, so I have removed `testifylint` for now.

**Comment by patrickhuie19** [comment]
Does this make #246 obsolete? I had a feeling something like this was coming so I didn't merge

File: .golangci.yml
Comment (jmank88) [inline .golangci.yml:1]: - https://github.com/mdempsky/unconvert
- https://github.com/ryanrolds/sqlclosecheck
- https://github.com/sonatard/noctx
- https://github.com/ultraware/whitespace

[revive](https://github.com/mgechev/revive):
 - https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#exported
 - https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#var-naming

File: pkg/loop/internal/broker.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (a *atomicBroker) DialWithOptions(id uint32, opts ...grpc.DialOption) (conn
 	return a.load().DialWithOptions(id, opts...)
 }
 
-func (a *atomicBroker) NextId() uint32 {
+func (a *atomicBroker) NextId() uint32 { //nolint:revive
```
Comment (patrickhuie19) [inline pkg/loop/internal/broker.go:42]: what does this do?

File: pkg/monitoring/poller.go [hunk 2]
```diff
@@ -75,11 +75,10 @@ func (s *sourcePoller) Run(ctx context.Context) {
 					continue
 				} else if errors.Is(err, context.Canceled) {
 					return
-				} else {
-					s.log.Errorw("failed to fetch from source", "error", err)
-					reusedTimer.Reset(s.pollInterval)
-					continue
 				}
+				s.log.Errorw("failed to fetch from source", "error", err)
```
Comment (patrickhuie19) [inline pkg/monitoring/poller.go:79]: This feels odd to me - the previous formatting was easier to read

**Comment by jmank88** [comment]
> Does this make #246 obsolete? I had a feeling something like this was coming so I didn't merge

No, this doesn't affect CI other than the version bump.

**Review by patrickhuie19 — CHANGES_REQUESTED** [review]
> Does this make #246 obsolete? I had a feeling something like this was coming so I didn't merge

If it does, I think this PR is a good opportunity to update the sonar properties:
```
sonar.projectName=chainlink-common
```

File: pkg/loop/internal/broker.go (hunk 1 above)
Comment (jmank88) [reply pkg/loop/internal/broker.go:42]: Ignores the naming failure on `Id` instead of `ID`, since we are implementing an interface from another package.

**Review by patrickhuie19 — APPROVED** [review]

File: pkg/monitoring/poller.go (hunk 2 above)
Comment (jmank88) [reply pkg/monitoring/poller.go:79]: The idea is to reduce unnecessary indentation.
https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#superfluous-else

Comment (jmank88) [reply pkg/monitoring/poller.go:79]: https://go.dev/doc/effective_go#control-structures
> In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.

**Pull request merged**